src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferStreamAsyncMemoryOverloads.cs

```diff
@@ -0,0 +1,299 @@
+            if (undesiredReadAsyncMethod == null)
+            {
+                return;
+            }
+
+            IMethodSymbol? undesiredWriteAsyncMethod = writeAsyncMethodGroup.GetFirstOrDefaultMemberWithParameterInfos(undesiredParameters);
+            if (undesiredWriteAsyncMethod == null)
+            {
+                return;
+            }
+
+            IMethodSymbol? undesiredReadAsyncMethodWithCancellationToken = readAsyncMethodGroup.GetFirstOrDefaultMemberWithParameterInfos(undesiredParametersWithCancellationToken);
+            if (undesiredReadAsyncMethodWithCancellationToken == null)
+            {
+                return;
+            }
+
+            IMethodSymbol? undesiredWriteAsyncMethodWithCancellationToken = writeAsyncMethodGroup.GetFirstOrDefaultMemberWithParameterInfos(undesiredParametersWithCancellationToken);
+            if (undesiredWriteAsyncMethodWithCancellationToken == null)
+            {
+                return;
+            }
+
+            // Create the arrays with the exact parameter order of the undesired methods
+            var preferredReadAsyncParameters = new[]
+            {
+                ParameterInfo.GetParameterInfo(readOnlyMemoryType),  // ReadOnlyMemory<byte> buffer
+                ParameterInfo.GetParameterInfo(cancellationTokenType), // CancellationToken
+            };
+
+            var preferredWriteAsyncParameters = new[]
+            {
+                ParameterInfo.GetParameterInfo(memoryType),  // ReadOnlyMemory<byte> buffer
+                ParameterInfo.GetParameterInfo(cancellationTokenType), // CancellationToken
+            };
+
+            // Retrieve the preferred methods, which are used for constructing the rule message
+            IMethodSymbol? preferredReadAsyncMethod = readAsyncMethodGroup.FirstOrDefault(x =>
+                x.Parameters.Count() == 2 && x.Parameters[0].Type is INamedTypeSymbol type && type.ConstructedFrom.Equals(memoryType));
+            if (preferredReadAsyncMethod == null)
+            {
+                return;
+            }
+
+            IMethodSymbol? preferredWriteAsyncMethod = writeAsyncMethodGroup.FirstOrDefault(x =>
+                x.Parameters.Count() == 2 && x.Parameters[0].Type is INamedTypeSymbol type && type.ConstructedFrom.Equals(readOnlyMemoryType));
+            if (preferredWriteAsyncMethod == null)
+            {
+                return;
+            }
+
+            context.RegisterOperationAction(context =>
+            {
+                IInvocationOperation invocation = (IInvocationOperation)context.Operation;
+
+                // Only accept these two cases:
+                // - await {WriteAsync()|ReadAsync()}
+                // - await {WriteAsync()|ReadAsync()}.ConfigureAwait()
+                if ((invocation.Parent != null && invocation.Parent.Kind == OperationKind.Await) ||
+                        (invocation.Parent!.Parent != null && invocation.Parent!.Parent.Kind == OperationKind.Await &&
```

Comment:
If invocation.Parent is null this is going to throw.

I'd precede this whole if with `if (invocation.Parent == null) { return; }`, and perhaps save invocation.Parent to a local, since you use it a lot.

Reply:
Thanks. Done.